oops/oops_test.go

```diff
@@ -898,6 +898,67 @@ subsequent stack frames truncated
 	}
 }
 
+func TestCollectMetadata(t *testing.T) {
+	err := errors.New("oops")
+	err1 := oops.WrapfWithMetadata(err, map[string]interface{}{
+		"a": "b",
+	}, "err1")
+	err2 := oops.WrapfWithMetadata(err1, map[string]interface{}{
+		"c": 1,
+	}, "err2")
+	err3 := oops.WrapfWithMetadata(err1, map[string]interface{}{
+		"a": 1,
+		"d": "e",
+	}, "err3")
+	nonOopsErr := fmt.Errorf("wraps oops err: %w", err1)
+
+	testCases := []struct {
+		description string
+		error       error
+		want        map[string]interface{}
+	}{
+		{
+			description: "metadata is retrieved from oops error",
+			error:       err1,
+			want: map[string]interface{}{
+				"a": "b",
+			},
+		},
+		{
+			description: "metadata is aggregated from 2 oops errors",
+			error:       err2,
+			want: map[string]interface{}{
+				"a": "b",
+				"c": 1,
+			},
+		},
+		{
+			description: "metadata is aggregated from 2 oops errors and key deduplicated by selecting inner most value",
```

Comment:
I think it would make more sense to use the outer-most value? WDYT?

Reply:
I thought of returning the inner most because that's the root cause of the error but I'm not too opinionated about that tbh. I think the better option might be to include both and rename the key when duplicates are found but didn't wanna add the additional complexity unless is something we really think valuable. 
For now I'm happy to select the outer most: https://github.com/samsarahq/go/pull/32/commits/d8b91cfab71ba05798f3996252d97f811de9f71a